### PR TITLE
VDS-2326 Add query for loading country list

### PR DESCRIPTION
**Pull request by trueboroda**

### CODE REVIEW

**Review by asvishnyakov — APPROVED** [review]

**Review by pushnitsa — APPROVED** [review]

File: samples/VirtoCommerce.Exp.ExtensionSamples/VirtoCommerce.Exp.ExtensionSamples.csproj [hunk 1]
```diff
@@ -14,7 +14,7 @@
         <PackageReference Include="AutoMapper.Extensions.Microsoft.DependencyInjection" Version="8.0.1" />
         <PackageReference Include="MediatR" Version="8.0.1" />
         <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="3.1.8" />
-        <PackageReference Include="VirtoCommerce.Platform.Core" Version="3.57.0" />
+        <PackageReference Include="VirtoCommerce.Platform.Core" Version="3.71.0" />
```
Comment (pushnitsa) [inline samples/VirtoCommerce.Exp.ExtensionSamples/VirtoCommerce.Exp.ExtensionSamples.csproj:17]: Downgrade platform version till minimum with ICountryService and raise the platform dependency in module.manifest

Comment (trueboroda) [reply samples/VirtoCommerce.Exp.ExtensionSamples/VirtoCommerce.Exp.ExtensionSamples.csproj]: To 3.66 was downgraded.

File: samples/VirtoCommerce.Exp.ExtensionSamples/VirtoCommerce.Exp.ExtensionSamples.csproj [hunk 2]
```diff
@@ -14,7 +14,7 @@
         <PackageReference Include="AutoMapper.Extensions.Microsoft.DependencyInjection" Version="8.0.1" />
         <PackageReference Include="MediatR" Version="8.0.1" />
         <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="3.1.8" />
-        <PackageReference Include="VirtoCommerce.Platform.Core" Version="3.57.0" />
+        <PackageReference Include="VirtoCommerce.Platform.Core" Version="3.66.0" />
```
Comment (Vectorfield4) [inline samples/VirtoCommerce.Exp.ExtensionSamples/VirtoCommerce.Exp.ExtensionSamples.csproj:17]: Update `<PackageReference Include="VirtoCommerce.Platform.Security" Version="3.57.0" />` to `<PackageReference Include="VirtoCommerce.Platform.Security" Version="3.66.0" />`.

File: src/VirtoCommerce.ExperienceApiModule.Core/Queries/GetCountriesQueryHandler.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+using System.Threading.Tasks;
+using VirtoCommerce.ExperienceApiModule.Core.Infrastructure;
+using VirtoCommerce.Platform.Core.Common;
+
+namespace VirtoCommerce.ExperienceApiModule.Core.Queries
+{
+    public class GetCountriesQueryHandler : IQueryHandler<GetCountriesQuery, GetCountriesResponse>,
+        IQueryHandler<GetRegionsQuery, GetRegionsResponse>
+
+
```
Comment (Vectorfield4) [inline src/VirtoCommerce.ExperienceApiModule.Core/Queries/GetCountriesQueryHandler.cs:11]: Remove empty lines.

File: src/VirtoCommerce.ExperienceApiModule.Core/Queries/GetCountriesQueryHandler.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    {
+        private readonly ICountriesService _countriesService;
+
+        public GetCountriesQueryHandler(ICountriesService countriesService)
+        {
+            _countriesService = countriesService;
+        }
+        public async Task<GetCountriesResponse> Handle(GetCountriesQuery request, CancellationToken cancellationToken)
```
Comment (Vectorfield4) [inline src/VirtoCommerce.ExperienceApiModule.Core/Queries/GetCountriesQueryHandler.cs:19]: Add line and XML-doc for this method.

File: src/VirtoCommerce.ExperienceApiModule.Core/Queries/GetCountriesQueryHandler.cs [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        {
+            var countries = await _countriesService.GetCountriesAsync();
+            var result = new GetCountriesResponse { Countries = countries };
+
+            return result;
+        }
+
+        public async Task<GetRegionsResponse> Handle(GetRegionsQuery request, CancellationToken cancellationToken)
```
Comment (Vectorfield4) [inline src/VirtoCommerce.ExperienceApiModule.Core/Queries/GetCountriesQueryHandler.cs:27]: Add XML-doc for this method, please.

File: src/VirtoCommerce.ExperienceApiModule.Core/Schemas/CountryRegionType.cs [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+using VirtoCommerce.Platform.Core.Common;
+
+namespace VirtoCommerce.ExperienceApiModule.Core.Schemas
+{
+    public class CountryRegionType : ObjectGraphType<CountryRegion>
+    {
+        public CountryRegionType()
+        {
+            Field(x => x.Id);
+            Field(x => x.Name);
```
Comment (Vectorfield4) [inline src/VirtoCommerce.ExperienceApiModule.Core/Schemas/CountryRegionType.cs:11]: Add descriptions, please. [Example](https://github.com/VirtoCommerce/vc-module-experience-api/blob/dev/src/VirtoCommerce.ExperienceApiModule.Core/Schemas/CurrencyType.cs#L10).

File: src/VirtoCommerce.ExperienceApiModule.Core/VirtoCommerce.ExperienceApiModule.Core.csproj [hunk 7]
```diff
@@ -31,7 +31,7 @@
         <PackageReference Include="VirtoCommerce.CoreModule.Core" Version="3.2.0" />
         <PackageReference Include="VirtoCommerce.TaxModule.Core" Version="3.2.0" />
         <PackageReference Include="VirtoCommerce.SearchModule.Core" Version="3.13.0" />
-        <PackageReference Include="VirtoCommerce.Platform.Core" Version="3.57.0" />
+        <PackageReference Include="VirtoCommerce.Platform.Core" Version="3.66.0" />
```
Comment (Vectorfield4) [inline src/VirtoCommerce.ExperienceApiModule.Core/VirtoCommerce.ExperienceApiModule.Core.csproj:34]: Update `<PackageReference Include="VirtoCommerce.Platform.Security" Version="3.57.0" />` to `<PackageReference Include="VirtoCommerce.Platform.Security" Version="3.66.0" />`.

**Review by Vectorfield4 — CHANGES_REQUESTED** [review]

**Review by Vectorfield4 — APPROVED** [review]

**Pull request merged**